tests/waku_archive/archive_utils.nim

```diff
@@ -23,26 +23,11 @@ proc newSqliteArchiveDriver*(): ArchiveDriver =
 proc newWakuArchive*(driver: ArchiveDriver): WakuArchive =
   WakuArchive.new(driver).get()
 
-proc computeArchiveCursor*(
-    pubsubTopic: PubsubTopic, message: WakuMessage
-): ArchiveCursor =
-  ArchiveCursor(
-    pubsubTopic: pubsubTopic,
-    senderTime: message.timestamp,
-    storeTime: message.timestamp,
-    digest: computeDigest(message),
-    hash: computeMessageHash(pubsubTopic, message),
-  )
-
 proc put*(
     driver: ArchiveDriver, pubsubTopic: PubSubTopic, msgList: seq[WakuMessage]
 ): ArchiveDriver =
   for msg in msgList:
-    let
-      msgDigest = computeDigest(msg)
-      msgHash = computeMessageHash(pubsubTopic, msg)
-      _ = waitFor driver.put(pubsubTopic, msg, msgDigest, msgHash, msg.timestamp)
-        # discard crashes
+    let _ = waitFor driver.put(computeMessageHash(pubsubTopic, msg), pubsubTopic, msg)
```

Comment:
Maybe a future simplification would be the following and then `computeMessageHash` within the `put` method.
```suggestion
    let _ = waitFor driver.put(pubsubTopic, msg)
```

Reply:
Yes we could do that.

Another solution would be to refactor our internal WakuMessage type to include the hash and compute it only once at creation.